### ISSUE TITLE
Underwater sound effect doesn't play continuously

### ISSUE THREAD

**Issue by walkawayy** [rr-/Tomb1Main#344]
The underwater sound effect plays on repeat now thanks to #306. However, there is a slight delay between the track ending and starting again. I'm not sure if this is something in the game code or just a bad sample. Also the sample has like white noise in the background that isn't present in TR2. So I think the sample is just bad quality. Ideally we'd have a better sample but we'd have to possibly need to do something like #157. Another solution could be to have a high quality music track and play it that way instead of as a sound effect, but that's not distributed with TombATI which is how most people get the music files.

Here's TR1's noise:
[sfx_underwater.zip](https://github.com/rr-/Tomb1Main/files/7877840/sfx_underwater.zip)
 
Here's TR2's version:
[tr2-underwater-sfx.zip](https://github.com/rr-/Tomb1Main/files/7877890/tr2-underwater-sfx.zip)

**Comment by Richard-L** [comment]
The delay in the loop is the same on PS1. The sound stops before it repeats. The sample just isn't seamless. Personally I'm not sure I think this _has_ to be fixed, but I can understand why you posted this.

Does it make sense to tell the code to loop the sample x amount of ms earlier before each repetition ends? I wonder if you could fine tune it like that without adding new files altogether. Else I would honestly just leave it.

**Comment by walkawayy** [comment]
Thanks for confirming, I didn't know that. So maybe this can be a feature instead of a bug.

I'm not sure if there's a way to check what time the current track is in. I couldn't find anything in the code to do that.

**Comment by walkawayy** [comment]
It would have to be a hack like the secret one checking for track number most likely. Until the day the custom sound effects possibly happens.

**Comment by walkawayy** [comment]
Took a super ugly, hacky approach to possibly solve this until injecting a new UNDERWATER_SFX is possible. It's ugly and the audio clicks/pops before the sound restarts (but that can happen if the camera switches between underwater/land mode too). I don't want to spend too much time on something that we don't want to actually implement, so @rr- let me know if this is even something we want to consider.

https://github.com/walkawayy/Tomb1Main/commit/d71b607aa7ee6f4ceb6ded1e2e66cd927af61e48

**Comment by rr-** [comment]
No, it's not really acceptable. Can you check for silent samples like I proposed? A sketch of a patch to get you started:

```diff
diff --git a/src/specific/s_audio_sample.c b/src/specific/s_audio_sample.c
index 2d763889..7a214619 100644
--- a/src/specific/s_audio_sample.c
+++ b/src/specific/s_audio_sample.c
@@ -10,6 +10,8 @@
 #include <libavformat/avformat.h>
 #include <libswresample/swresample.h>
 
+#define SILENT_SAMPLE 0.01
+
 typedef struct AUDIO_SAMPLE {
     float *sample_data;
     int channels;
@@ -280,6 +282,13 @@ static bool S_Audio_SampleLoad(int sample_id, const char *content, size_t size)
     sample->channels = swr.src_channels;
     sample->sample_data = working_buffer;
 
+    // trim silence at the end of the samples
+    while (sample->num_samples > 0
+           && fabsf(sample->sample_data[sample->num_samples - 1])
+               < SILENT_SAMPLE) {
+        sample->num_samples--;
+    }
+
     return true;
 
 fail:
``

**Comment by Richard-L** [comment]
Honestly I would just leave this as is and not change it 🤷‍♂️

**Comment by walkawayy** [comment]
Ah sample_data I screwed up and was checking only sample_data[0] so I was wondering why it wasn't working.
